library/Fission/Types.hs

```diff
@@ -36,42 +38,65 @@ import           Fission.Platform.Heroku.AddOn
 
 import qualified Fission.Web.Auth              as Auth
 import           Fission.Web.Server.Reflective
-import qualified Fission.Web.Types             as Web
 
-instance Has Web.Host cfg => MonadReflectiveServer (RIO cfg) where
-  getHost = Config.get
+-- | The top-level app type
+newtype Fission a = Fission { unFission :: RIO Config a }
+  deriving newtype ( Functor
+                   , Applicative
+                   , Monad
+                   , MonadIO
+                   , MonadUnliftIO
+                   , MonadReader Config
+                   , MonadThrow
+                   , MonadCatch
+                   , MonadMask
+                   )
+
+-- TODO move to Fisson.hs
+runFission :: MonadIO m => Config -> Fission a -> m a
+runFission cfg actions =
+  actions
+    |> unFission
+    |> runRIO cfg
+
+instance MonadLogger Fission where
+  monadLoggerLog loc src lvl msg = Fission (monadLoggerLog loc src lvl msg)
+
+instance MonadTime Fission where
+  currentTime = liftIO getCurrentTime
+
+instance MonadReflectiveServer Fission where
+  getHost = asks host
```

Comment:
Question: Where does this asks function come from?

Reply:
piggybacking off this, is there any type-checking going on with `asks`? or are we saying "trust me, there's a value in the config with key `host`"?

Reply:
@bechurch 

> Question: Where does this asks function come from?

`asks` comes from `Reader`, which is what `RIO` uses to automagically thread the config through. It takes a function, and plugs the config into it, and gives you the result. Good for things like selectors (as seen here).

Here's some docs: https://www.stackage.org/haddock/lts-14.19/rio-0.1.12.0/RIO-Prelude.html#v:asks

Reply:
@dholms 

> piggybacking off this, is there any type-checking going on with asks? or are we saying "trust me, there's a value in the config with key host"?

Haskell doesn't really have a "trust me" switch... it's more like "prove it to me!".

Because we have so much abstraction down in the business logic, we can reasonably hardcode the `Config` in `Fission`. It knows that we have `host` in scope because the type of Fission is `RIO Config`, and `Config` has a `host` field.